evap/staff/tests/test_views.py

```diff
@@ -1653,6 +1653,73 @@ def test_evaluation_copy(self):
         self.assertEqual(copied_evaluation.contributions.count(), 4)
 
 
+class TestCourseCopyView(WebTestStaffMode):
+    url = "/staff/semester/1/course/1/copy"
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.manager = make_manager()
+        cls.semester = baker.make(Semester, pk=1)
+        cls.other_semester = baker.make(Semester, pk=2)
```

Comment:
please don't use explicit pks to avoid database integrity errors and create the url like in, e.g., `TestEvaluationCopyView`

Reply:
Interesting issue here
Run `./manage.py test evap.staff.tests.test_views.TestCourseCopyView`. You will get an IntegrityError on
```
form.submit().follow()
```
in line 1706. This is triggered by the CourseCopyForm doing
```
new_course: Course = super.save()
```
in staff/forms.py:327

In super.save(), it will do `self.instance.save()`, which fails because it violates the unique constraint on the course primary key. However, `self.instance` correctly does not have a pk set, so we're just trying to use the key the database gives us here.

Seems to me like the auto-counter of the db is at 1 and it doesn't check whether that key can actually be used.

Reply:
minimally reproducible in `./manage.py shell_plus`:
```python
>>> from model_bakery import baker
>>> s = baker.make(Semester)
>>> ct = baker.make(CourseType)
>>>
>>> c1 = baker.make(Course, semester=s, type=ct)
>>> c1.pk
144
>>> # So next ID would be 145. Mess it up with
>>> c2 = baker.make(Course, semester=s, type=ct, pk=145)
>>> c2.pk
145
>>> c3 = baker.prepare(Course, semester=s, type=ct)
>>> print(c3.pk)
None
>>> 
>>> c3.save()  # boom
```

Reply:
sorry, i missed this. If a test uses only hard-coded non-conflicting IDs, all should be fine. A test never specifying pks should also be fine. hard-coded pks at all in a test should also be fine. Different tests shouldn't affect one another. I imagine that only mixing both methods within a test, like in your repro case, would break.